site/content/docs/main/csi-snapshot-data-movement.md

```diff
@@ -0,0 +1,390 @@
+
+- Velero built-in data mover creates a volume with the same specification of the source volume.  
+- Velero built-in data mover waits Kubernetes to provision the volume, this may take some time varying from storage providers, but if the provision cannot be finished in a given time, Velero built-in data mover will cancel this `DataDownload` CR. The timeout is configurable through the same node-agent's parameter `data-mover-prepare-timeout`.  
+- After the volume is provisioned, Velero built-in data mover starts to transfer the data from the backup storage according to the backup storage location defined by users.  
+- When the data transfer completes or any error happens, Velero built-in data mover set the `DataDownload` CR to the terminal state, either `Completed` or `Failed`.  
+- Velero built-in data mover also monitors the cancellation request to the `DataDownload` CR, once that happens, it cancel its ongoing activities, clean up the intermediate resources and set the `DataDownload` CR to `Cancelled`.  
+
+
+### Prallelism
+
+Velero calls the CSI plugin concurrently for the volume, so `DataUpload`/`DataDownload` CRs are created concurrently by the CSI plugin. For more details about the call between Velero and CSI plugin, check the [Volume Snapshot Data Movement design][1].  
+In which manner the `DataUpload`/`DataDownload` CRs are processed is totally decided by the data mover you select for the backup/restore.  
+
+For Velero built-in data mover, it uses Kubernetes' scheduler to mount a snapshot volume/restore volume associated to a `DataUpload`/`DataDownload` CR into a specific node, and then the `DataUpload`/`DataDownload` controller (in node-agent daemonset) in that node will handle the `DataUpload`/`DataDownload`.  
+At present, a `DataUpload`/`DataDownload` controller in one node handles one request at a time.  
+That is to say, the snapshot volumes/restore volumes may spread in different nodes, then their associated `DataUpload`/`DataDownload` CRs will be processed in parallel; while for the snapshot volumes/restore volumes in the same node, their associated `DataUpload`/`DataDownload` CRs are processed sequentially.  
+
+You can check in which node the `DataUpload`/`DataDownload` CRs are processed and their prallelism by watching the `DataUpload`/`DataDownload` CRs:
+
+```bash
+kubectl -n velero get datauploads -l velero.io/backup-name=YOUR_BACKUP_NAME -w
+```
+
+```bash
+kubectl -n velero get datadownloads -l velero.io/restore-name=YOUR_RESTORE_NAME -w
+```
+
+### Cancellation
+
+At present, Velero backup and restore doesn't support end to end cancellation that is launched by users.  
+However, Velero cancels the `DataUpload`/`DataDownload` in below scenarios automatically:
+- When Velero server is restarted
+- When node-agent is restarted  
+- When an ongoing backup/restore is deleted
```

Comment:
one more scenario is the item operation timeout, right ?

Reply:
Added, thanks.